File: app/src/main/java/com/doapps/habits/helper/ConnectionManager.java
package com.doapps.habits.helper;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class ConnectionManager {

    /**
     * @return true if user is connected to Internet
     */
    public static boolean isConnected(Context context){
        ConnectivityManager connectivityManager
                = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }
}

File: app/src/main/java/com/doapps/habits/helper/ConnectionManager.kt
package com.doapps.habits.helper

import android.content.Context
import android.net.ConnectivityManager

object ConnectionManager {

  /**
   * @return true if user is connected to Internet
   */
  @JvmStatic
  fun isConnected(context: Context): Boolean {
    val connectivityManager =
        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
    val activeNetworkInfo = connectivityManager.activeNetworkInfo
    return activeNetworkInfo != null && activeNetworkInfo.isConnected
  }
}
